src/Brush.elm

```diff
@@ -0,0 +1,1102 @@
+        )
+    , custom "touchmove"
+        (D.andThen
+            (\touches ->
+                case touches of
+                    { position } :: _ ->
+                        D.succeed
+                            { message = tagger (MouseMove position)
+                            , stopPropagation = True
+                            , preventDefault = True
+                            }
+
+                    [] ->
+                        D.fail ""
+            )
+            Events.decodeTouches
+        )
+    , custom "touchend"
+        (D.succeed
+            { message = tagger MouseUp
+            , stopPropagation = True
+            , preventDefault = True
+            }
+        )
+    ]
+        ++ (case drag of
+                Nothing ->
+                    [ custom "mousedown"
+                        (D.map4
+                            (\pos meta alt shift ->
+                                let
+                                    el =
+                                        if keysEnabled && meta then
+                                            OverlayElement
+
+                                        else
+                                            element
+
+                                    mode =
+                                        if el == SelectionElement then
+                                            Drag
+
+                                        else if keysEnabled && alt then
+                                            Center
+
+                                        else
+                                            Handle
+                                in
+                                { message = tagger (MouseDown mode (keysEnabled && shift) element pos)
+                                , stopPropagation = True
+                                , preventDefault = False
+                                }
+                            )
+                            Events.decodeMousePosition
+                            (D.field "metaKey" D.bool)
+                            (D.field "altKey" D.bool)
+                            (D.field "shiftKey" D.bool)
+                        )
+                    ]
+
+                Just _ ->
+                    []
+           )
+
+
+{-| Initializes a brush that allows brusing in the X axis.
+-}
+initX : Extent -> Brush OneDimensional
+initX =
+    init True False
+
+
+{-| Initializes a brush that allows brusing in the Y axis.
+-}
+initY : Extent -> Brush OneDimensional
+initY =
+    init False True
+
+
+{-| Initializes a two dimensional brush.
+-}
+initXY : Extent -> Brush TwoDimensional
+initXY =
+    init True True
+
+
+init : Bool -> Bool -> Extent -> Brush dimension
+init x y extent =
+    Brush { drag = Nothing, keysEnabled = True, shifting = False, x = x, y = y, selection = Nothing, extent = extent }
+
+
+{-| By default the brush will use the meta/alt and shift keys to change behavior. You can disable this with this function.
```

Comment:
The initial introduction to the Brush feature only mentions the `Meta` and `Alt` keys. Does `Shift` do something as well? I may be getting confused with key names...

Reply:
I think the problem here is the other way around. Shift works fine (it locks the selection in one dimension), but I don't think `Meta` does anything at the moment.

Reply:
I reread everything and I see that it's confusing. I added a bit about shift to the first paragraph.